models/m_payment_mollie_api.erl

```diff
@@ -195,40 +221,42 @@ fetch_payments_since(Oldest, NextLink, Acc, Context) ->
             Error
     end.
 
-handle_missed_recurrent(#{ <<"recurringType">> := <<"recurring">> } = JSON, Context) ->
+handle_missed_recurrent(#{ <<"sequenceType">> := <<"recurring">> } = JSON, Context) ->
     #{
         <<"id">> := ExtId,
-        <<"links">> := #{
-            <<"webhookUrl">> := WebhookUrl
-        }
+        <<"webhookUrl">> := WebhookUrl
     } = JSON,
     [ WebhookUrl1 | _ ] = binary:split(WebhookUrl, <<"?">>),
-    PaymentNr = lists:last( binary:split(WebhookUrl1, <<"/">>, [ global ])),
-    case m_payment:get(PaymentNr, Context) of
-        {ok, Payment} ->
-            case proplists:get_value(psp_module, Payment) of
-                mod_payment_mollie ->
+    FirstPaymentNr = lists:last( binary:split(WebhookUrl1, <<"/">>, [ global ])),
+    case m_payment:get(FirstPaymentNr, Context) of
+        {ok, FirstPayment} ->
+            % This is the original payment starting the sequence
+            case proplists:lookup(psp_module, FirstPayment) of
+                {psp_module, mod_payment_mollie} ->
                     % Fetch the status from Mollie
-                    PaymentId = proplists:get_value(id, Payment),
+                    {id, FirstPaymentId} = proplists:lookup(id, FirstPayment),
                     m_payment_log:log(
-                        PaymentId,
+                        FirstPaymentId,
                         <<"WEBHOOK">>,
                         [
                            {psp_module, mod_payment_mollie},
                            {description, "New webhook payment info"},
                            {payment, JSON}
                         ],
                         Context),
-                    handle_new_payment(PaymentId, Payment, JSON, Context);
-                PSP ->
-                    lager:error("Payment PSP Mollie webhook call for unknown PSP ~p / ~p: ~p", [PaymentNr, ExtId, PSP]),
+                    handle_new_payment(FirstPaymentId, FirstPayment, JSON, Context);
+                {psp_module, PSP} ->
+                    lager:error("Payment PSP Mollie webhook call for unknown PSP ~p / ~p: ~p",
+                                [FirstPaymentNr, ExtId, PSP]),
```

Comment:
This probably intentionally crashes when no psp_module is found in the Payment - would indeed be very strange if it wasn't at this point.

Reply:
Indeed, also a column in the db.